File: python/21-30/Merge Two Sorted Lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    # @param two ListNodes
    # @return a ListNode
    def mergeTwoLists(self, l1, l2):
        if not l1:
            return l2
        if not l2:
            return l1
        su = ListNode(0)
        head = su
        while l1!=None and l2!=None:
            if l1.val <l2.val:
                head.next = l1
                l1 = l1.next
            else:
                head.next = l2
                l2 =l2.next
            head = head.next
            head.next=None
        if l1!=None:
            head.next = l1
        else:
            head.next = l2
        return su.next

File: Cpp/21-30/Merge Two Sorted Lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
 
 /*
    runtime:8ms
    error:0
 */
class Solution {
public:
    ListNode* mergeTwoLists(ListNode* l1, ListNode* l2) {
        if(l1==NULL) return l2;
        if(l2==NULL) return l1;
        ListNode head(0),*temp=&head;
        while(l1!=NULL&&l2!=NULL){
            if(l1->val<l2->val){
                temp= temp->next = l1;
                l1 = l1.next;
                temp->next=NULL;
            }else{
                temp=temp->next=l2;
                l2=l2->next;
                temp->next=NULL;
            }
        }if(l1==NULL) temp->next = l2;
        else temp->next = l1;
        return head.next;
    }
};
